src/legacy/core_plugins/data/public/expressions/lib/loader.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Ast } from '@kbn/interpreter/common';
+import { Observable, Subject } from 'rxjs';
+import { first, share } from 'rxjs/operators';
+import { Adapters, InspectorSession } from '../../../../../../plugins/inspector/public';
+import { execute, ExpressionDataHandler } from './execute';
+import { ExpressionRenderHandler } from './render';
+import { RenderId, Data, IExpressionLoaderParams } from './_types';
+import { getInspector } from '../services';
+
+export class ExpressionLoader {
+  data$: Observable<Data>;
+  update$: Observable<any>;
+  render$: Observable<RenderId>;
+  events$: Observable<any>;
+
+  private dataHandler: ExpressionDataHandler;
+  private renderHandler: ExpressionRenderHandler;
+  private dataSubject: Subject<Data>;
+  private data: Data;
+
+  constructor(element: HTMLElement, expression: string | Ast, params: IExpressionLoaderParams) {
+    this.dataSubject = new Subject();
+    this.data$ = this.dataSubject.asObservable().pipe(share());
+
+    this.renderHandler = new ExpressionRenderHandler(element);
+    this.render$ = this.renderHandler.render$;
+    this.update$ = this.renderHandler.update$;
+    this.events$ = this.renderHandler.events$;
```

Comment:
I'm not sure why set all these observables on this object, we can already access them through `this.renderHandler`.

Reply:
renderHandler is private and not accessible to the outside